### ISSUE TITLE
Give users some idea of what `func build` is doing

### ISSUE THREAD

**Issue by omerbensaadon** [knative-sandbox/kn-plugin-func#555]
# Context
When we interviewed users, many of them watched `func build` hang for quite awhile... some (at least one) mentioned that they would probably cancel the process altogether given how long it took. 

This can probably be delivered by setting `func build` to run in verbose by default, but **a better user experience would be aligning to something similar to what `kn quickstart` does** which is to create some output for each smaller step within the process so users know the process is still running.

# Story 

**As Deandra the Developer**
**I want** to know what actions `func build` is performing under the hood
**So that** I better understand what `func build` actually does 
**And so that** I have some idea of how long it will take 

# Acceptance Criteria

- [ ] Give me some idea of what `func build` is doing
```
GIVEN I have already run func create myfunc && cd myfunc 
WHEN I run func build 
THEN I have some general idea of what func build is doing under the hood
```

**Comment by omerbensaadon** [comment]
this story was originally AC#2 for #552

**Comment by lance** [comment]
I'm going to push back on this a bit - we already have the `-v` flag which pipes the buildpack build output to the terminal, making it pretty clear exactly what is happening under the hood, and without the `-v` flag, I think the CLI should be as quiet as possible. Finally, the implementation of this is unclear. What exactly is meant by "some idea of what's happening under the hood"? There is really just one thing happening, and that's that the builder/buildpacks are being applied to the project. Any more granular than that, and we're basically back at using the `-v` flag.

**Comment by omerbensaadon** [comment]
I think ideally, a download progress bar and then a message to the effect of "building" after that would satisfy this story... 

That being said, Naina and I agree that #537 will resolve our most pressing concerns for the time-being.

**Comment by lance** [comment]
> I think ideally, a download progress bar and then a message to the effect of "building" after that would satisfy this story...

I'm going to continue pushing back on this, because aside from the download progress bar, this is what we have (plus what was added in #537). And it's not clear what a download progress bar would be measuring. Download of what?

**Comment by lkingland** [comment]
> without the -v flag, I think the CLI should be as quiet as possible

I completely agree with this.  However, I also understand that on longer tasks it is preferable to have some kind of progress indication to assure the user that something is happening.

Early on, we identified this as a potential issue, and added the "spinner" with a periodic update.  Later, we added even more user updates as Lance mentions in #537.  If I am remembering correctly, #537 was not part of the user interviews.

I would hazard a guess that we are getting close to "enough notifications to assure the user it's not hanged, just slow".  Of course, if this proves not to be enough, we can add even more.

I would suggest that #537 actuall fixes this issue (this patch was not part of the sustem when running the interviews)

**Comment by lance** [comment]
> I would suggest that #537 actually fixes this issue (this patch was not part of the system when running the interviews)

I agree. I suspect that @omerbensaadon has a full plate with his new work at GitHub, and while I don't want to preemptively close this without some acknowledgement from Omer, I believe this issue has been resolved in #537 and can be closed. Omer, feel free to reopen with specific requests for functional changes if you have them, or create a new issue.
